src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -710,18 +710,17 @@ protected virtual void Dispose(bool disposing)
         {
             foreach (var componentState in _componentStateById.Values)
             {
-                if (componentState.Component is IDisposable disposable)
+                try
+                {
+                    componentState.Dispose();
+                }
+                catch (Exception exception)
                 {
-                    try
-                    {
-                        disposable.Dispose();
```

Comment:
This is not the only place where components get disposed. This code path is what happens when the renderer itself is disposed (which disposes all remaining components in it), but we also dispose components individually when they get removed from their parent. We'll need some more logic in `ComponentState.DisposeInBatch` to ensure it disposes the entire `ComponentState` now.

Reply:
Unrelated, but should Dispose not Dispose the item if `_componentWasDisposed` was set to true in `DisposeInBatch`?

Reply:
We definitely shouldn't double-dispose components. My guess is that we wouldn't anyway because if `DisposeInBatch` was called, then we'd have removed that `ComponentState` from the renderer at the same time, at least in normal cases.

But I'm not against being clearer about this in the code by having `ComponentState` track whether or not it was already disposed so it definitely doesn't double-do it.